### DISCUSSION TITLE
This file is not part the project...

### DISCUSSION THREAD

**Discussion by Ateros4me** [google/oboe#1443] [Q&A] [answered]
For some reason, when I click on the native-lib.cpp file in the cpp directory within Android Studio the bar saying

_This file is not part of the project. Please include it in the build file (build.gradle, CMakeList.txt..._

However, I have this in the CMakeLists.txt file

add_library( # Sets the name of the library.
        native-lib
        # Sets the library as a shared library.
        SHARED
        # Provides a relative path to your source file(s).
        src/main/cpp/native-lib.cpp
        src/main/cpp/DuplexEngine.cpp
        )

So, why does it think it's not part of the project when it is added as a library element?

**Comment by Ateros4me** [comment]
You may as well close this as i’ve given up with OBOE as i just can’t get it to work

**Comment by philburk** [comment, marked as answer]
You also have to tell Gradle to look for a CMake file.  Did you include this in the build.gradle file?

    externalNativeBuild {
        cmake {
            path 'src/main/cpp/CMakeLists.txt'
        }
    }

For [example](https://github.com/google/oboe/blob/9a9487a82a771ea8a7705861129a65df3da147c1/samples/drumthumper/build.gradle#L30), from DrumThumper.

**Reply by Ateros4me** [reply]
Thanks Phil, after adding the above another problem has occurred within the CMakeList.txt file but i’ll submit that as a separate discussion.
